internal/transport/test/upload_pack.go

```diff
@@ -28,263 +28,270 @@ type UploadPackSuite struct {
-	s.Equal(ar1, ar2)
+	s.Require().NoError(err)
+	s.Require().Equal(ar1, ar2)
 }
 
 func (s *UploadPackSuite) TestDefaultBranch() {
+	ctx := context.TODO()
 	r, err := s.Client.NewSession(s.Storer, s.Endpoint, s.EmptyAuth)
-	s.NoError(err)
-	conn, err := r.Handshake(context.TODO(), transport.UploadPackService)
-	s.NoError(err)
-	defer func() { s.Nil(conn.Close()) }()
-
-	info, err := conn.GetRemoteRefs(context.TODO())
-	s.NoError(err)
-	s.NotNil(info)
+	s.Require().NoError(err)
+	conn, err := r.Handshake(ctx, transport.UploadPackService)
+	s.Require().NoError(err)
+	defer func() { s.Require().Nil(conn.Close()) }()
+
+	info, err := conn.GetRemoteRefs(ctx)
+	s.Require().NoError(err)
+	s.Require().NotNil(info)
 	symrefs := conn.Capabilities().Get(capability.SymRef)
-	s.Len(symrefs, 1)
-	s.Equal("HEAD:refs/heads/master", symrefs[0])
+	s.Require().Len(symrefs, 1)
+	s.Require().Equal("HEAD:refs/heads/master", symrefs[0])
 }
 
 func (s *UploadPackSuite) TestAdvertisedReferencesFilterUnsupported() {
 	r, err := s.Client.NewSession(s.Storer, s.Endpoint, s.EmptyAuth)
-	s.NoError(err)
+	s.Require().NoError(err)
 	conn, err := r.Handshake(context.TODO(), transport.UploadPackService)
-	s.NoError(err)
-	defer func() { s.Nil(conn.Close()) }()
+	s.Require().NoError(err)
+	defer func() { s.Require().Nil(conn.Close()) }()
 
 	info, err := conn.GetRemoteRefs(context.TODO())
-	s.NoError(err)
-	s.NotNil(info)
-	s.True(conn.Capabilities().Supports(capability.MultiACK))
+	s.Require().NoError(err)
+	s.Require().NotNil(info)
+	s.Require().True(conn.Capabilities().Supports(capability.MultiACK))
 }
 
 func (s *UploadPackSuite) TestCapabilities() {
 	r, err := s.Client.NewSession(s.Storer, s.Endpoint, s.EmptyAuth)
-	s.NoError(err)
+	s.Require().NoError(err)
 	conn, err := r.Handshake(context.TODO(), transport.UploadPackService)
-	s.NoError(err)
-	defer func() { s.Nil(conn.Close()) }()
+	s.Require().NoError(err)
+	defer func() { s.Require().Nil(conn.Close()) }()
 
 	info, err := conn.GetRemoteRefs(context.TODO())
-	s.NoError(err)
-	s.NotNil(info)
-	s.Len(conn.Capabilities().Get(capability.Agent), 1)
+	s.Require().NoError(err)
+	s.Require().NotNil(info)
+	s.Require().Len(conn.Capabilities().Get(capability.Agent), 1)
 }
 
 func (s *UploadPackSuite) TestUploadPack() {
 	r, err := s.Client.NewSession(s.Storer, s.Endpoint, s.EmptyAuth)
-	s.NoError(err)
+	s.Require().NoError(err)
 	conn, err := r.Handshake(context.TODO(), transport.UploadPackService)
-	s.NoError(err)
-	defer func() { s.Nil(conn.Close()) }()
+	s.Require().NoError(err)
+	defer func() { s.Require().Nil(conn.Close()) }()
 
+	beforeCount := s.coundObjects(s.Storer)
 	req := &transport.FetchRequest{}
 	req.Wants = append(req.Wants, plumbing.NewHash("6ecf0ef2c2dffb796033e5a02219af86ec6584e5"))
 
 	err = conn.Fetch(context.Background(), req)
-	s.NoError(err)
+	s.Require().NoError(err)
+
+	afterCount := s.coundObjects(s.Storer)
 
-	s.checkObjectNumber(s.Storer, 28)
+	s.Require().Equal(28, afterCount-beforeCount)
```

Comment:
The `checkObjectNumber` func is no longer used, so given that the new `coundObjects` has largely the same scope, I'd say we better off removing the former from the code base.